src/rez/utils/pip.py

```diff
@@ -0,0 +1,705 @@
+"""
+Python packaging related utilities.
+"""
+import os.path
+import sys
+from email.parser import Parser
+import platform
+import subprocess
+import re
+
+import pkg_resources
+
+from rez.vendor.distlib.util import parse_name_and_version
+from rez.vendor.packaging.version import (
+    parse as packaging_parse,
+    LegacyVersion as packaging_LegacyVersion,
+    InvalidVersion as packaging_InvalidVersion
+)
+from rez.vendor.packaging.requirements import Requirement as packaging_Requirement
+from rez.vendor.version.requirement import Requirement
+from rez.vendor.version.version import Version, VersionRange
+
+from rez.utils.logging_ import print_warning
+from rez.exceptions import PackageRequestError
+from rez.system import System
+
+
+def get_pip_version(pip_exe):
+    """Get the version of the given pip executable.
+
+    Returns:
+        str: Pip version, eg "19.1.1", or None if version couldn't be detected
+    """
+    pattern = r"pip\s(?P<ver>\d+\.*\d*\.*\d*)"
+
+    if "Windows" in platform.system():
+        # https://github.com/nerdvegas/rez/pull/659
+        ver_str = subprocess.check_output(
+            pip_exe + " -V",
```

Comment:
We might want to do `python -m pip` instead. That way we could be sure which pip will be used. I would even go one step further and use `sys.executable` to get the path to the correct python.

Reply:
This code needs to be updated actually.. currently it causes pip to be run from outside of its rez context (if rezified pip was found), which is completely wrong. I'm gonna fix this quite soon so ignore this for now. I'll make it part of this PR though.

On that note, sys.executable would only get used if no rezified pip packages was found (but yes I agree that it should be used in that case, and that pip exe shouldn't be used directly).